ucb-rit/coldfront
# Support on-the-fly configuration changes; display deployment-specific User Agreement Form

Refs #383

**Changes**
- Included [`django-constance`](https://django-constance.readthedocs.io/en/latest/), which allows settings to be stored in Redis and updated live.
    - Included configuration of the module in the Ansible settings template.
    - Added a context processor that makes these configuration variables universally available across templates.
    - Used this to configure the launch date of the portal and the link to the feedback Google Form, which appear in the alert on the home page.
- Displayed the correct cluster user access agreement form based on the current instance.
    - Hid the SU usage pop quiz for LRC.
- Updated `README.md` to include notes on `django-constance`.

**How to Test**
- Review the code changes and add comments as needed.
- Ensure that the test suite passes.
    - This appears to be failing on an LRC VM for reasons that should be resolved in #428.
        - Some date comparison failures in `test_allocation_renewal_processing_runner` (when run at certain times) are also resolved there.
- Manual Testing
    - Re-run the Ansible playbook.
    - As a superuser on a BRC VM, navigate to `/admin/constance/config/`. Set:
        - `FEEDBACK_FORM_URL`: https://docs.google.com/forms/d/e/1FAIpQLScZoYXkFMAVEYhoqE2guG64JQb4dsPcwidlhZ0W-TZ2t_Va4w/viewform
        - `LAUNCH_DATE`: 2021-06-01.
    - On the home page, ensure that the values above are used.
    - Navigate to the User Access Agreement.
        - Ensure that the correct text is rendered for the current instance.
        - As a user who has not signed the agreement, ensure that the pop quiz only appears for a BRC instance.
        - To switch between instances, update `dev_settings.FLAGS` to enable one flag and disable the other. (This will likely break other functionality, but should work for testing this specific functionality.)
